### PR TITLE
feat: [IOBP-647] Latest used method pre-selection

### PR DESCRIPTION
**Pull request by Hantex9**
## Short description
This PR implements the logic to pre-select a payment method from the user wallets that have been recently used and will be shown in the appropriate “Recently Used” list item

## List of changes proposed in this pull request
- Added an additional selector to filter the recently used payment method from the user wallet with the field `updateDate` as more recent;
- Added a pre-selection statement into the reducer when the action `paymentsGetPaymentUserMethodsAction.success` is fired to filter the most recent payment method.

## How to test
Start a new payment flow with the new payment section feature flag enabled;
When you start a payment, you should be able to see the most recently used payment method pre-selected

## Preview
https://github.com/pagopa/io-app/assets/34343582/38e455e8-3e10-450d-a360-8c4024d3a75b

### CODE REVIEW

File: locales/en/index.yml [hunk 1]
```diff
@@ -1695,6 +1695,7 @@ wallet:
       header: Seleziona un metodo
       yourMethods: Saved
       otherMethods: All methods
+      latestMethod: Latest used
```
Comment (thisisjp) [inline locales/en/index.yml:1698]:
```suggestion
      latestMethod: Recently used
```

**Review by thisisjp — APPROVED** [review]

File: ts/features/payments/checkout/components/CheckoutPaymentMethodsList.tsx [hunk 2]
```diff
@@ -37,6 +38,9 @@ const CheckoutPaymentMethodsList = () => {
   const paymentAmountPot = useIOSelector(walletPaymentAmountSelector);
   const allPaymentMethods = useIOSelector(walletPaymentAllMethodsSelector);
   const userWallets = useIOSelector(walletPaymentUserWalletsSelector);
+  const latestPaymentUsed = useIOSelector(
```
Comment (mastro993) [inline ts/features/payments/checkout/components/CheckoutPaymentMethodsList.tsx:41]:
```suggestion
  const latestPaymentMethodUsed = useIOSelector(
```

File: ts/features/payments/checkout/components/CheckoutPaymentMethodsList.tsx [hunk 3]
```diff
@@ -51,6 +55,19 @@ const CheckoutPaymentMethodsList = () => {
...
 
+  const latestPaymentMethodListItem = useMemo(
+    () =>
+      pipe(
+        latestPaymentUsed,
+        O.map(mapUserWalletToRadioItem),
+        O.map(Array.of),
+        O.map(A.map(O.fromNullable)),
+        O.map(A.compact),
+        O.getOrElse(() => [] as Array<RadioItem<string>>)
```
Comment (mastro993) [inline ts/features/payments/checkout/components/CheckoutPaymentMethodsList.tsx:66]:
```suggestion
        latestPaymentUsed,
        O.chainNullableK(mapUserWalletToRadioItem),
        O.map(A.of),
        O.getOrElse(() => [] as Array<RadioItem<string>>)
```

This is clearer, what do you think?

File: ts/features/payments/checkout/components/CheckoutPaymentMethodsList.tsx (hunk 2 above)
Comment (Hantex9) [reply ts/features/payments/checkout/components/CheckoutPaymentMethodsList.tsx]: Addressed into 4e22fe4be0d5c23ab4d99c02c31cc5a9c8c42763

File: ts/features/payments/checkout/components/CheckoutPaymentMethodsList.tsx (hunk 3 above)
Comment (Hantex9) [reply ts/features/payments/checkout/components/CheckoutPaymentMethodsList.tsx]: You're an fp-ts wizard! 🧙‍♂️

Addressed into 4e22fe4be0d5c23ab4d99c02c31cc5a9c8c42763

**Review by mastro993 — APPROVED** [review]
LGTM

**Pull request merged**